docs/topics/db/transactions.txt

```diff
@@ -252,6 +252,130 @@ by Django or by third-party libraries. Thus, this is best used in situations
+be discarded and never called.
+
+
+Order of execution
+------------------
+
+On-commit functions for a given transaction are executed in the order they were
+registered.
+
+
+Exception handling
+------------------
+
+If one on-commit hook within a given transaction raises an uncaught exception,
+no later-registered hooks in that same transaction will run. (This is, of
+course, the same behavior as if you'd executed the hooks sequentially yourself
+without :func:`on_commit`.)
+
+
+Timing of execution
+-------------------
+
+Your hook functions are executed *after* a successful commit, so if they fail,
+it will not cause the transaction to roll back. They are executed conditionally
+upon the success of the transaction, but they are not *part* of the
+transaction. For the intended use cases (mail notifications, Celery tasks,
+etc), this is probably fine. If it's not (if your follow-up action is so
+critical that its failure should mean the failure of the transaction itself),
+then you don't want to use the :func:`on_commit` hook. (Instead, you may want
+`two-phase commit`_.)
+
+.. _two-phase commit: http://en.wikipedia.org/wiki/Two-phase_commit_protocol
+
+Use in tests
+------------
+
+Django's :class:`~django.test.TestCase` class wraps each test in a transaction
+and rolls back that transaction after each test, in order to provide test
+isolation. This means that no transaction is ever actually committed, thus your
+:func:`on_commit` hooks will never be run. If you need to test the results of
+an :func:`on_commit` hook, you may need to use
+:class:`~django.test.TransactionTestCase` instead.
+
+
+Savepoints
+----------
+
+Savepoints (i.e. nested :func:`atomic` blocks) are handled correctly. That is,
+an :func:`on_commit` hook registered after a savepoint (in a nested
+:func:`atomic` block) will be called after the outer transaction is committed,
+but not if a rollback to that savepoint or any previous savepoint occurred
+during the transaction.
+
+Use autocommit and transaction.atomic
+-------------------------------------
+
+.. warning::
+    On-commit hooks only works correctly in :ref:`autocommit
+    mode<managing-autocommit>` or with the :func:`atomic` /
+    :setting:`ATOMIC_REQUESTS <DATABASE-ATOMIC_REQUESTS>` transaction API. If
+    you set disable autocommit on your connection and/or use lower-level
+    transaction APIs directly, the :func:`on_commit` hook will not work.
+
+Commit hooks are not run until autocommit is restored on the connection
+following the commit (because otherwise any queries done in a commit hook would
+open an implicit transaction, preventing the connection from going back into
+autocommit mode).
+
+When in autocommit mode and outside of an :func:`atomic` block, the hook will run
+immediately, not on commit.
+
+Why no rollback hook?
+---------------------
+
+A rollback hook is harder to implement robustly than a commit hook, since a
+variety of things can cause an implicit rollback. For instance, your database
```

Comment:
More lazy writing on my part. Change to "For instance, if your database connection is dropped because your process was killed without a chance to shut down gracefully, your rollback hook will never run."